File: GatewayUI/react-ui/src/app/models/publication.ts
export interface Publication {
	id: number;
	title: string;
	authors: string;
	journal: string;
	publishedAt: Date;
	doi: string;
	citation: string | null;
	scaffoldGroupIds: number[];
	descriptorTypeIds: number[];
}

export interface PublicationToCreate {
	title: string;
	authors: string;
	journal: string;
	publishedAt: string;
	doi: string;
	citation: string | null;
}

export interface DescriptorRuleToCreate {
	descriptorTypeId: number;
	jobMode: number;
	jobId: string | null;
}

File: KnowledgeBaseBackend/Data/Models/Publication.cs
namespace KnowledgeBaseApi.Data.Models
{
	public class Publication
	{
		public int Id { get; set; }
		public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
		public string Title { get; set; } = null!;
		public string Authors { get; set; } = null!;
		public string Journal { get; set; } = null!;
		public string Doi { get; set; } = null!;
		public DateTime PublishedAt { get; set; }
		public string? Citation { get; set; }
	}
}
